src/Storages/ExternalStream/Kafka/Kafka.h

```diff
@@ -36,37 +60,52 @@ class Kafka final : public StorageExternalStreamImpl
     SinkToStoragePtr write(const ASTPtr & query, const StorageMetadataPtr & metadata_snapshot, ContextPtr context) override;
 
     bool produceOneMessagePerRow() const { return settings->one_message_per_row; }
+    Int32 topicRefreshIntervalMs() const { return topic_refresh_interval_ms; }
     const String & brokers() const { return settings->brokers.value; }
     const String & dataFormat() const override { return data_format; }
-    const String & topic() const { return settings->topic.value; }
-    const klog::KConfParams & properties() const { return kafka_properties; }
-    const klog::KafkaWALAuth & auth() const noexcept { return *auth_info; }
+    const String & topicName() const { return settings->topic.value; }
     const ASTPtr & shardingExprAst() const { assert(!engine_args.empty()); return engine_args[0]; }
     bool hasCustomShardingExpr() const;
-    klog::KafkaWALSimpleConsumerPtr getConsumer(int32_t fetch_wait_max_ms = 200) const;
+
+    RdKafka::Producer & getProducer();
+    RdKafka::Topic & getProducerTopic();
+
+    RdKafka::ConsumerPool::Entry getConsumer() const
+    {
+        assert(consumer_pool);
+        return consumer_pool->get(/*max_wait_ms=*/1000);
```

Comment:
add more log for consumer_pool (such as current pool size)

Reply:
Every time when a new consumer is created, it will be logged and it will log the number of consumer ( consumers are named in the format of `consumer-<number-of-consumer>` ). For now, I don't think we need any more logs.